File: Unit - 1/Brute Force Strategy/C implementation/stringMatching.c
//Program to implement brute-force string matching
#include<stdio.h>
#include<string.h>
#define SIZE 50

int stringMatch(char text[], char pattern[])
{
    int n = strlen(text);
    int m = strlen(pattern);
    int i, j;
    for(i = 0; i <= n - m; i++){
        j = 0;
        while(j < m && text[i + j] == pattern[j])
        {
            j++;
        }
        if(j == m)
            return i;
    }
    return -1;
}

int main()
{
    char text[SIZE], pattern[SIZE];
    printf("Enter the text: ");
    scanf("%s", text);
    printf("Enter the pattern: ");
    scanf("%s", pattern);
    int index = stringMatch(text, pattern);
    if(index == -1)
    {
        printf("String match not found\n");
    }
    else
    {
        printf("String match found at index %d\n", index);
    }
}

File: Unit - 1/Brute Force Strategy/Java Implementation/StringMatching.java
//Program to implement brute-force string matching
public class StringMatching {
    
    private static int stringMatch(String text, String pattern){
        for(int i = 0;  i <= text.length() - pattern.length(); i++){
            int j = 0;
            while(j < pattern.length() && text.charAt(i + j) == pattern.charAt(j)){
                j++;
            }
            if(j == pattern.length()){
                return i;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        String text = "NOBODY_NOTICED_HIM";
        String pattern = "NOT";
        int index = stringMatch(text, pattern);
        if(index == -1){
            System.out.println("Pattern not found in the text");
        } else {
            System.out.println("Pattern found at index: " + index);
        }
    }
}
